stubs/docutils/docutils/nodes.pyi

```diff
@@ -178,11 +179,66 @@ class Labeled: ...
 
 # Root Element
 
+_Document: TypeAlias = document
+_Decoration: TypeAlias = decoration
+
 class document(Root, Structural, Element):
-    parent: None
+    current_source: str
+    current_line: int | None
+    settings: Values
+    reporter: Reporter
+    indirect_targets: list[target]
+    substitution_defs: dict[str, substitution_definition]
+    substitution_names: dict[str, str]
+    refnames: dict[str, list[Element]]
+    refids: dict[str, list[Element]]
+    nameids: dict[str, str]
+    nametypes: dict[str, bool]
+    ids: dict[str, Element]
+    footnote_refs: dict[str, list[footnote_reference]]
+    citation_refs: dict[str, list[citation_reference]]
+    autofootnotes: list[footnote]
+    autofootnote_refs: list[footnote_reference]
+    symbol_footnotes: list[footnote]
+    symbol_footnote_refs: list[footnote_reference]
+    footnotes: list[footnote]
+    citations: list[citation]
+    autofootnote_start: int
+    symbol_footnote_start: int
+    id_start: int
+    parse_messages: list[system_message]
+    transform_messages: list[system_message]
     transformer: Transformer
-    def __init__(self, settings, reporter: Reporter, rawsource: str = "", *children: Node, **attributes) -> None: ...
-    def __getattr__(self, __name: str) -> Incomplete: ...
+    decoration: decoration
+    document: document
+    def __init__(self, settings: Values, reporter: Reporter, *args: Node, **kwargs: Any) -> None: ...
+    def asdom(self, dom: Any = None) -> Any: ...
```

Comment:
Functionally the same, but since `None` is explicitly checked it might be nice to document that `None` is a valid argument:

```suggestion
    def asdom(self, dom: Any | None = None) -> Any: ...
```